### DISCUSSION TITLE
Entity could not be found in cache for entity_id:

### DISCUSSION THREAD

**Discussion by gcortes** [zachowj/node-red-contrib-home-assistant-websocket#631] [Q&A] [answered]
I'm getting the error: Entity could not be found in cache for entity_id: several times a day in my debug log. I'm not sure if the error is a problem with this project or Home Assistant. The node causing the error and the error details are shown here:

<img width="816" alt="2022-04-22_07-54-10" src="https://user-images.githubusercontent.com/5271010/164741251-166518eb-8212-42cb-8941-ed5361373c33.png">

The entry for the boolean in Home Assistant is shown here:

<img width="392" alt="2022-04-22_07-55-01" src="https://user-images.githubusercontent.com/5271010/164741416-ef741508-6773-4af8-b2cf-74c9b324e088.png">

I have another boolean of the same type that works fine. I'm running version 0.43.1 of this project.

**Comment by zachowj** [comment, marked as answer]
I would guess that the current-state is being triggered before you have connected to Home Assistant.

**Reply by gcortes** [reply]
That might be it. The nodes that are throwing the error are all connected to individual five minute timers. I hadn't seen any errors in about a day when I started working on some flows. I did a deploy a few times without issue. The last deploy I did, which involved changes to a number of nodes, caused the errors to appear. Maybe Node-Red reconnects to home assistant when I deploy is activated.

I can live with the error, but would like to get rid of it. Is there a way, after the timer triggers, to test if home assistant is available?

It happened again, but with only three nodes. I changed the timers to "inject once after 2 seconds, then" instead of 0.1 seconds. No errors when I deployed it.

It's been a couple of days with no errors. Setting the delay on the timers did the trick.
